packages/perseus/src/widgets/expression/expression.test.tsx

```diff
@@ -550,78 +550,231 @@ describe("Expression Widget", function () {
+                times: false,
+                functions: [],
+                extraKeys: ["x"],
+            });
+
+            // Assert
+            expect(result.keypadConfiguration.extraKeys).toEqual(["x"]);
+        });
     });
 
-    it("should handle scientific button set", async () => {
-        // Arrange
-        // Act
-        // Assert
-        expect(
-            keypadConfigurationForProps({
-                answerForms: [],
-                buttonSets: ["scientific"],
+    describe("interactive: full vs answerless", () => {
+        beforeAll(() => {
+            registerAllWidgetsForTesting();
+        });
+
+        let userEvent: UserEvent;
+        beforeEach(() => {
+            userEvent = userEventLib.setup({
+                advanceTimers: jest.advanceTimersByTime,
+            });
+
+            jest.spyOn(Dependencies, "getDependencies").mockReturnValue(
+                testDependencies,
+            );
+
+            // Mocked for loading graphie in svg-image
+            global.fetch = jest.fn(() =>
+                Promise.resolve({
+                    text: () => "",
+                    ok: true,
+                }),
+            ) as jest.Mock;
+        });
+
+        it("is answerable with full widget options", async () => {
+            // Arrange
+            const expressionOptions: PerseusExpressionWidgetOptions = {
+                answerForms: [
+                    {
+                        considered: "correct",
+                        form: true,
+                        simplify: false,
+                        value: "1i",
+                    },
+                ],
+                buttonSets: [],
+                times: false,
+                functions: [],
+                extraKeys: ["i"],
+            };
+
+            const fullItem: PerseusRenderer = {
+                content: "[[☃ expression 1]]",
+                images: {},
+                widgets: {
+                    "expression 1": {
+                        type: "expression",
+                        version: {major: 2, minor: 0},
+                        options: expressionOptions,
+                    },
+                },
+            };
+
+            // Act
+            const {renderer} = renderQuestion(fullItem);
+
+            await userEvent.click(
+                screen.getByRole("button", {name: "open math keypad"}),
+            );
+            await userEvent.click(screen.getByRole("button", {name: "1"}));
+            await userEvent.click(screen.getByRole("tab", {name: "Extras"}));
+            await userEvent.click(screen.getByRole("button", {name: "i"}));
+            act(() => jest.runOnlyPendingTimers());
+
+            const userInput = renderer.getUserInputMap();
+            const score = scorePerseusItem(fullItem, userInput, "en");
+
+            // Assert
+            expect(score).toHaveBeenAnsweredCorrectly();
+        });
+
+        it("is interactive with answerless widget options", async () => {
```

Comment:
This is the important one. Without answer data we:

- Render
- Answer
- Send the input off for scoring